### ISSUE TITLE
Environment variables are overwritten when using multiple functions

### ISSUE THREAD

**Issue by mrkplt** [serverless/serverless-google-cloudfunctions#167]
Hello!

I am trying to deploy 6 cloud functions, each of which has a STATE_ID environment variable defined. What is happening is that each cloud function ends up with STATE_ID as "MD" which you will note is the last function defined. This doesn't seem right. 

Below there is a trimmed down copy of my serverless.yml

Is this a bug? It feels like a bug. My interpretation of how it should work just might be messed up. Any help would be appreciated!

Thanks, 
Mark

```service: mailingimport

plugins:
  - serverless-google-cloudfunctions

provider:
  name: google
  runtime: go111
  region: us-east1
  project: eng_project
  credentials: keyfile.json
  
package:
  exclude:
    - '**'
  include:
    - mailingimport.go

functions:
  # Pennsylvania
  mailing_import_pa:
    handler: MailingImportPa
    events:
      - event:
          eventType: providers/cloud.pubsub/eventTypes/topic.publish
          resource: projects/eng_project/topics/mailing-importer
    environment:
      STATE_ID: 'PA'
  # Colorado
  mailing_import_co:
    handler: MailingImportCo
    events:
      - event:
          eventType: providers/cloud.pubsub/eventTypes/topic.publish
          resource: projects/eng_project/topics/mailing-importer
    environment:
      STATE_ID: 'CO'
  # New York City
  mailing_import_nyc:
    handler: MailingImportNyc
    events:
      - event:
          eventType: providers/cloud.pubsub/eventTypes/topic.publish
          resource: projects/eng_project/topics/mailing-importer
    environment:
      STATE_ID: 'NYC'
  # South Carolina
  mailing_import_sc:
    handler: MailingImportSc
    events:
      - event:
          eventType: providers/cloud.pubsub/eventTypes/topic.publish
          resource: projects/eng_project/topics/mailing-importer
    environment:
      STATE_ID: 'SC'
  # North Carolina
  mailing_import_nc:
    handler: MailingImportNc
    events:
      - event:
          eventType: providers/cloud.pubsub/eventTypes/topic.publish
          resource: projects/eng_project/topics/mailing-importer
    environment:
      STATE_ID: 'NC'
  # Maryland
  mailing_import_md:
    handler: MailingImportMd
    events:
      - event:
          eventType: providers/cloud.pubsub/eventTypes/topic.publish
          resource: projects/eng_project/topics/mailing-importer
    environment:
      STATE_ID: 'MD'
```

**Comment by simonasdev** [comment]
fixed in #162

**Comment by mather** [comment]
Not merged yet :(
